optuna_dashboard/ts/components/GraphParetoFront.tsx

```diff
@@ -94,6 +94,69 @@ const filterFunc = (trial: Trial, directions: StudyDirection[]): boolean => {
   )
 }
 
+const makeHovertext = (trial: Trial): string => {
+  return JSON.stringify(
+    {
+      number: trial.number,
+      values: trial.values,
+      params: trial.params
+        .map((p) => [p.name, p.value])
+        .reduce((obj, [key, value]) => ({ ...obj, [key]: value }), {}),
+    },
+    undefined,
+    "  "
+  ).replace(/\n/g, "<br>")
+}
+
+const makeScatterObject = (
+  trials: Trial[],
+  objectiveXId: number,
+  objectiveYId: number,
+  hovertemplate: string,
+  dominated: boolean
+): Partial<plotly.PlotData> => {
+  const marker = makeMarker(trials, dominated)
+  return {
+    x: trials.map((t) => t.values![objectiveXId] as number),
+    y: trials.map((t) => t.values![objectiveYId] as number),
+    text: trials.map((t) => makeHovertext(t)),
+    mode: "markers",
+    hovertemplate: hovertemplate,
+    marker: marker,
+    showlegend: false,
+  }
+}
+
+const makeMarker = (
+  trials: Trial[],
+  dominated: boolean
+): Partial<plotly.PlotData> => {
+  if (dominated) {
+    return {
+      line: { width: 0.5, color: "Grey" },
+      // @ts-ignore
+      color: trials.map((t) => t.number),
+      colorscale: "Blues",
+      reversescale: true,
+      colorbar: {
+        title: "Trial",
+      },
+    }
+  } else {
+    return {
+      line: { width: 0.5, color: "Grey" },
+      // @ts-ignore
+      color: trials.map((t) => t.number),
+      colorscale: "Reds",
+      colorbar: {
+        title: "Best Trial",
+        x: 1.1,
+        xpad: 80,
```

Comment:
[nit] It looks that Optuna uses `xpad: 40`. Do you think we should use 80 here?

Reply:
@c-bata 
Yes. Otherwise, the red and blue bars will be overlapped.

Reply:
I see. Thanks. LGTM!